### ISSUE TITLE
Use real objects for getHost/getPeer

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7200]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#475|
|Type|enhancement|
|Created|2004-01-15 09:07:12Z|

Attachments:

* [peerhost.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6e7/6e75d23a0574afce62799aab4afcd882f53d6680/a40cef99264161aa0fd995731d772e65e8981ee6.patch) (4102 bytes) - added by exarkun on 2004-01-15 09:07:12Z - 
* [getHost.patch.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6e7/6e75d23a0574afce62799aab4afcd882f53d6680/e381806fb51f3b64bfa048eb1b0587c4a7ee828e.txt) (4776 bytes) - added by exarkun on 2004-01-18 01:58:26Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__475 475
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1074157632000000 1074157632000000
changetime__1470664916467872 1470664916467872
version__ 
owner__ 
cc__radix cc__exarkun cc__itamarst cc__dash
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Patch attached.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Users are going to want to use these. E.g., if isinstance(o, IPv4Address) might
htappen, people will want to see API doc, etc.. As such, I'd like it to be
somewhere other than base.py, which is mostly implementation specific stuff.
Other than that, need to add support for UNIXAddress and for UDP and TLS. But
the idea seems fine to me.

Oh, and bug in __eq__, it should be comparing self._bwHack, not self.type when
comparing to tuples. Also __str__ oughtn't pretend to be tuple, we don't want
people using the icky tuple API no more.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
ServerFactory.buildProtocol is currently invoked with (host, port). 
ClientFactory.buildProtocol is invoked with (transportIdentifier, host, port).

Do we want to preserve this behavior, or fix ServerFactory?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Both should get IPv4Address objects. The one for ServerFactory should get its
own backwards compatability hack for a 2-tuple. Won't break old code, and the
inconsistency is solved since people should use new API from now on anyway.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Slightly improved patch attached.  Still doesn't cover UNIX, UDP, or TLS.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
isn't this done now?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|GitHub commented|
|-|-|

In changeset 39c506ffcc09ec5ce8764ca8e669e02cdcee1848
```
#!CommitTicketReference repository="" revision="39c506ffcc09ec5ce8764ca8e669e02cdcee1848"
Merge pull request #475 from twisted/8732-rodrigc-dict-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8732

Fix twisted.test.test_dict on Python 3
```
